web/src/lib/roadmap/exportRoadmapAsPdf.ts

```diff
@@ -0,0 +1,84 @@
+import html2canvas from "html2canvas";
+import jsPDF from "jspdf";
+
+const splitElementIntoImages = async (
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  element: any,
+  pageWidthInPx: number,
+  pageHeightInPx: number
+): Promise<{ totalHeight: number; canvas: string }[]> => {
+  const scaleFactor = pageWidthInPx / element.offsetWidth;
+  const totalHeight = element.offsetHeight / scaleFactor;
+  const pages = Math.ceil(totalHeight / pageHeightInPx);
+  const carryOverPixels = 25;
+  const images = [];
+
+  const clone = element.cloneNode(true);
+  clone.style.width = `${pageWidthInPx}px`;
+  clone.style.transformOrigin = "top left";
+  clone.style.position = "fixed";
+  clone.style.top = "100%";
+  clone.style.left = 0;
+  document.body.append(clone);
+
+  for (let i = 0; i < pages; i++) {
+    const result = { totalHeight, canvas: "" };
+    const canvas = await html2canvas(clone, {
+      width: pageWidthInPx,
+      height: Math.floor(Math.min(totalHeight - i * pageHeightInPx, pageHeightInPx)),
+      y: i === 0 ? 0 : i * pageHeightInPx - carryOverPixels,
+      windowWidth: Math.ceil(Math.max(element.offsetHeight, pageWidthInPx)),
+    });
+    result.canvas = canvas.toDataURL("image/png");
+
+    images.push(result);
+  }
+  clone.remove();
+  return images;
+};
+
+export const exportComponentsAsPDF = async (
+  componentIds: string[],
+  prePdfProcessingCb: () => void,
+  postPdfProcessingCb: () => void
```

Comment:
[pebble] two notes
- I think `Cb` means "callback" here, I don't think that's a common abbreviation (took me a min) and might be clearer to just write out
- With so many arguments to this function, especially with the same type signature, I think this would be clearer to use if we did named parameters instead